winlocalprocessspawner/winlocalprocessspawner.py

```diff
@@ -68,12 +68,8 @@ def get_env(self):
         except Exception as exc:
             self.log.warning("Failed to load user environment for %s: %s", self.user.name, exc)
         else:
-            # If the user profile is loaded, adjust APPDATA so the jupyter runtime files are stored
-            # in a per-user location.
-            if 'APPDATA' in user_env:
-                env['APPDATA'] = user_env['APPDATA']
-                env['USERPROFILE'] = user_env['USERPROFILE']
-            else:
+            env.update(user_env)
+            if not 'APPDATA' in user_env:

```

Comment:
We need to provide a sane value for TEMP, if there is no token. Maybe something like C:\Users\All Users\AppData\Local\Temp? other suggestions? A comment specifying that the if means no user profile could help future edits too...

Reply:
On Win 7 and Server 2008, there's no `C:\Users\All Users` and `C:\Users\Public` does not have AppData. I guess for this use case of no token, we can keep using the parent's value for TEMP.
I will restore TEMP in win_keep. When there's a valid token, that value will get overwritten with the value from user_env.

Reply:
Done, please check the latest version